src/librustc_lint/builtin.rs

```diff
@@ -1174,16 +1174,22 @@ fn check_const(cx: &LateContext<'_, '_>, body_id: hir::BodyId) {
 
 impl<'a, 'tcx> LateLintPass<'a, 'tcx> for UnusedBrokenConst {
     fn check_item(&mut self, cx: &LateContext<'_, '_>, it: &hir::Item<'_>) {
+        trace!("UnusedBrokenConst: check_item(_, _, {:?}", it);
         match it.kind {
-            hir::ItemKind::Const(_, body_id) => {
-                check_const(cx, body_id);
-            }
-            hir::ItemKind::Static(_, _, body_id) => {
+            hir::ItemKind::Const(_, body_id) | hir::ItemKind::Static(_, _, body_id) => {
+                trace!("UnusedBrokenConst: checking {:?}", body_id);
                 check_const(cx, body_id);
             }
             _ => {}
         }
     }
+    fn check_impl_item(&mut self, cx: &LateContext<'_, '_>, it: &hir::ImplItem<'_>) {
```

Comment:
You can also have associated constants with default values in trait definitions. Also consider const generics here, e.g. `foo::<{BAD_CONSTANT}>();`.

Reply:
in order to do this you can eval all anonconsts, which will also cover a few other cases (though I think they are already evaluated elsewhere)

Reply:
Let's do this in a separate PR though

Reply:
Sure thing